### ISSUE TITLE
predict result doesn't change with fobj each iter

### ISSUE THREAD

**Issue by modkzs** [Microsoft/LightGBM#267]
I try to use lightgbm with following fobj:
```
    def fairobj(preds, dtrain):
        labels = dtrain.get_label()
        con = 2
        x = -2 * labels / ((labels + preds) * (labels + preds))
        grad = con * x / (abs(x) + con)
        hess = con * con / (abs(x) + con) ** 2
        return grad, hess
```

but predict result doesn't change each iter. I try to print grad and hess, none of them is 0. 

when I use this fobj with init_model, it runs well.

**Comment by guolinke** [comment]
@modkzs 
Did you mean the training loss is always same with iteration goes?
Can you also paste full code?

**Comment by modkzs** [comment]
@guolinke Yes, training loss is same. So I compare predict result each iter, and it doesn't  change, too. The full code it very long, so I can paste code related with lgb:

```
def fairobj(preds, dtrain):
    labels = dtrain.get_label()
    con = 2
    x = -2 * labels / ((labels + preds) * (labels + preds))
    grad = con * x / (abs(x) + con)
    hess = con * con / (abs(x) + con) ** 2

    return grad, hess


def xg_eval_mae(yhat, dtrain):
    y = dtrain.get_label()
    return 'error', np.abs((yhat - y) / (yhat + y)).sum(), False

model = lgb.train(params, dtrain, num_boost_round=100000, valid_sets=d_valid, feval=xg_eval_mae, fobj=fairobj, early_stopping_rounds=50)
```

I try many params but result doesn't change. So it seems useless to paste params. 
PS: this fobj and fevl can act will with xgboost

**Comment by wxchan** [comment]
log can be controlled by:
1. 'verbose' in params, c++ log, -1 disables all
2. verbose_eval in train(), python log, False disables all

can you also paste your log of lightgbm and xgboost so we can see the difference? you might provide a sample data to reproduce such issue if possible.

btw, is this fairobj same as built-in fair objective?

**Comment by msn27149m** [comment]
is it multi-classification problem or just binary classification problem?
